tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFile.java

```diff
@@ -1462,22 +1463,70 @@ public static Set<String> getMatchingXMLFiles(File sourceDirs[], Matcher m) {
         return s;
     }
 
-    @Override
-    public Iterator<String> iterator() {
+    private final boolean DEFAULT_ITERATION_INCLUDES_EXTRAS = true;
+
+    public Iterator<String> iteratorDefault() {
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            return Iterators.filter(fullIterable().iterator(), p -> getStringValue(p) != null);
+        } else {
+            return iteratorWithoutExtras();
+        }
+    }
+
+    public Iterator<String> iteratorDefault(String prefix) {
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            if (prefix == null || prefix.isEmpty()) {
+                return iteratorDefault();
+            }
+            return Iterators.filter(iteratorDefault(), p -> p.startsWith(prefix));
+        } else {
+            return iteratorWithoutExtras(prefix);
+        }
+    }
+
+    public Iterator<String> iteratorDefault(Matcher pathFilter) {
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            if (pathFilter == null) {
+                return iteratorDefault();
+            }
+            return Iterators.filter(iteratorDefault(), p -> pathFilter.reset(p).matches());
+        } else {
+            return iteratorWithoutExtras(pathFilter);
+        }
+    }
+
+    public Iterator<String> iteratorDefault(String prefix, Comparator<String> comparator) {
+        if (comparator == null) {
+            throw new IllegalArgumentException("iteratorDefault requires non-null comparator");
+        }
+        if (DEFAULT_ITERATION_INCLUDES_EXTRAS) {
+            Iterator<String> it =
+                    (prefix == null || prefix.isEmpty())
+                            ? iteratorDefault()
+                            : iteratorDefault(prefix);
+            Set<String> orderedSet = new TreeSet<>(comparator);
+            it.forEachRemaining(orderedSet::add);
+            return orderedSet.iterator();
+        } else {
+            return iteratorWithoutExtras(prefix, comparator);
+        }
+    }
+
+    private Iterator<String> iteratorWithoutExtras() {
         return dataSource.iterator();
     }
 
-    public synchronized Iterator<String> iterator(String prefix) {
+    private synchronized Iterator<String> iteratorWithoutExtras(String prefix) {
         return dataSource.iterator(prefix);
```

Comment:
The original `iterator` methods are renamed to `iteratorWithoutExtras` and they are now private and rarely used if DEFAULT_ITERATION_INCLUDES_EXTRAS is true